action/execution.go

```diff
@@ -142,3 +172,60 @@ func (ex *Execution) SanityCheck() error {
 	}
 	return ex.AbstractAction.SanityCheck()
 }
+
+// SetEnvelopeContext sets the SealedEnvelope context to action context.
+func (ex *Execution) SetEnvelopeContext(selp SealedEnvelope) error {
+	if ex == nil {
+		return nil
+	}
+
+	if err := ex.AbstractAction.SetEnvelopeContext(selp); err != nil {
+		return err
+	}
+	if !ex.IsExternal() {
+		if ex.externChainID != 0 {
+			return errors.Errorf("invalid extern chain ID = %v", ex.externChainID)
+		}
+		return nil
+	}
+
+	// compute raw hash
+	var ethAddr *common.Address
+	if ex.contract != EmptyAddress {
+		addr, err := address.FromString(ex.contract)
+		if err != nil {
+			return errors.Wrapf(err, "invalid recipient address %s", ex.contract)
+		}
+		eAddr := common.BytesToAddress(addr.Bytes())
+		ethAddr = &eAddr
+	}
+	items := []interface{}{
```

Comment:
Dup code w/ https://github.com/iotexproject/iotex-core/pull/2586/files?file-filters%5B%5D=.go#diff-ede8b4d922fba992cf4cacfee119592fbac03586ade8fea4a8346866e4f2c07dR196 

Pls have one place to handle

Reply:
ok, will fix